README.md

```diff
@@ -1,2 +1,176 @@
+
+```erlang
+2> apns:connect(cert, my_first_connection).
+{ok,<0.87.0>}
+3> apns:connect(token, my_second_connection).
+{ok,<0.95.0>}
+```
+Note `cert` and `token` define the type we want.
+
+Although `Apns4erl` is supervising the connections `apns:connect/2` returns the connection `pid` just in case you wanted monitor it.
+
+## Push Notifications over `Provider Certificate` connections
+
+In order to send Notifications over `Provider Certificate` connection we will use `apns:push_notification/3,4`.
+
+We would need the connection, a notification, the device ID and some http2 headers. The connection is the `atom` we used when run `apns:connect/2` for setting a name, the device ID is provided by Apple, the notification is a `map` with the data we want to send, that map will be encoded to json later and the http2 headers can be explicitly sent as a parameter using `apns:push_notification/4` or can be defined at the `config` file, in that case we would use `apns:push_notification/3`.
+
+This is the `headers` format:
+
+```erlang
+-type headers()   :: #{ apns_id          => binary()
+                      , apns_expiration  => binary()
+                      , apns_priority    => binary()
+                      , apns_topic       => binary()
+                      , apns_collapse_id => binary()
+                      , apns_auth_token  => binary()
+                      }.
+```
+
+All of them are defined by Apple  [here](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html)
+
+Lets send a Notification.
+
+```erlang
+1> apns:connect(cert, my_first_connection).
+{ok,<0.85.0>}
+2> DeviceId = <<"a0dc63fb059cb9c13b03e5c974af3dd33d67fed4147da8c5ada0626439e18935">>.
+<<"a0dc63fb059cb9c13b03e5c974af3dd33d67fed4147da8c5ada0626439e18935">>
+3> Notification = #{aps => #{alert => <<"you have a message">>}}.
+#{aps => #{alert => <<"you have a message">>}}
+4> apns:push_notification(my_first_connection, DeviceId, Notification).
+{200,
+ [{<<"apns-id">>,<<"EFDE0D9D-F60C-30F4-3FF1-86F3B90BE434">>}],
+ no_body}
+```
+
+The result is the response itself, its format is:
+
+```erlang
+-type response()  :: { integer()          % HTTP2 Code
+                     , [term()]           % Response Headers
+                     , [term()] | no_body % Response Body
+                     } | timeout.
+```
+
+And that's all.
+
+## Push Notifications over `Provider Authentication Tokens` connections
+
+This is the other way APNs allows to send notifications. In this case we don't need a certificate but we will need a `p8` file with the private key we will use to sign the token. Lets suppose Apple provides us the file `APNsAuthKey_KEYID12345.p8`. We have to fill the `config` file key `token_keyfile` with the path to that file.
```

Comment:
allows -> allows you _or_ allows us
Lets suppose -> Lets assume
Apple provides us the file … -> we've got the file … from Apple
We have -> We then have